store/src/java/com/zimbra/cs/index/query/ItemQuery.java

```diff
@@ -37,17 +37,32 @@
 
     private final boolean isAllQuery;
     private final boolean isNoneQuery;
+    private final boolean isRangeQuery;
     private final List<ItemId> itemIds;
 
     public static Query create(Mailbox mbox, String str) throws ServiceException {
         boolean allQuery = false;
         boolean noneQuery = false;
+        boolean rangeQuery = false;
         List<ItemId> itemIds = new ArrayList<ItemId>();
 
         if (str.equalsIgnoreCase("all")) {
             allQuery = true;
         } else if (str.equalsIgnoreCase("none")) {
             noneQuery = true;
+        } else if(str.indexOf("--") > 0) {
+            String[] items = str.split("--");
+            for (int i = 0; i < items.length && i < 2; i++) {
```

Comment:
Does this mean that a query like "100--200--300" would be treated like a range from 100 to 200? Is it worth it to check for invalid range inputs like this?

Reply:
good catch, @ilyusha ! Yes, I should add a syntax check here.